diff --git a/osa2/puhelinluettelo/src/App.js b/osa2/puhelinluettelo/src/App.js
index 14e776d..99b70be 100644
--- a/osa2/puhelinluettelo/src/App.js
+++ b/osa2/puhelinluettelo/src/App.js
@@ -62,6 +62,12 @@ const App = () => {
         showMessage(`Added ${returnedPerson.name}`, setSuccessMessage)
         resetForm()
       })
+      .catch(error => {
+        console.log(error.response.data)
+        showMessage(
+          `${error.response.data.error}`,
+          setErrorMessage)
+      })
   }
 
   const filteredPersons = showName === '' ? persons :
